clients/unshaded/src/main/java/tachyon/client/file/AbstractTachyonFileSystem.java

```diff
@@ -72,14 +72,8 @@ public TachyonFile create(TachyonURI path, CreateOptions options)
     try {
       final long fileId = masterClient.create(path.getPath(), options);
       return new TachyonFile(fileId);
-    } catch (TachyonException e) {
-      if (e.getType() == TachyonExceptionType.BLOCK_INFO) {
-        throw new FileAlreadyExistsException(e.getMessage(), e);
-      } else {
-        TachyonException.unwrap(e, FileAlreadyExistsException.class);
-        TachyonException.unwrap(e, InvalidPathException.class);
-        throw e;
-      }
+    } catch (BlockInfoException e) {
```

Comment:
This exception means block size is invalid, so I think it should be propagate out directly instead of wrapping in `FileAlreadyExistsException`?